core/templates/dev/head/domain/exploration/exploration-id-validation.service.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to validate exploration id
+ */
+
+require('domain/summary/exploration-summary-backend-api.service.ts');
+
+angular.module('oppia').factory('ExplorationIdValidationService', [
+  'ExplorationSummaryBackendApiService',
+  function(ExplorationSummaryBackendApiService) {
+    return {
+      isExpIdValid: function(explorationId) {
```

Comment:
Think about naming a bit more. This is now a generic service that can be used by anything. So, what is "valid" in the general context? It's not clear.

Use a more specific name, something like: isExpPublished(explorationId).

Reply:
Done!